### PR TITLE
Reworked bmca running to use new interface.

### PR DESCRIPTION
**Pull request by davidv1992**

### CODE REVIEW

**Review by bramtweedegolf — APPROVED** [review]

**Review by folkertdev — APPROVED** [review]
again, some questions that I suspect are not blockers

File: statime/src/port/mod.rs [hunk 1]
```diff
@@ -236,15 +239,143 @@ impl Port<Running> {
...
+                parent_ds.parent_port_identity.clock_identity = defaultds.clock_identity;
+                parent_ds.parent_port_identity.port_number = 0;
+                parent_ds.grandmaster_identity = defaultds.clock_identity;
+                parent_ds.grandmaster_clock_quality = defaultds.clock_quality;
+                parent_ds.grandmaster_priority_1 = defaultds.priority_1;
+                parent_ds.grandmaster_priority_2 = defaultds.priority_2;
+
+                time_properties_ds.leap59 = false;
+                time_properties_ds.leap61 = false;
+                time_properties_ds.current_utc_offset = 37;
```
[the remark this answers is not in the document]
Comment (folkertdev) [reply statime/src/port/mod.rs:300]: what is this magic constant?

File: statime/src/port/mod.rs [hunk 2]
```diff
@@ -236,15 +239,143 @@ impl Port<Running> {
...
+
+                parent_ds.parent_port_identity.clock_identity = defaultds.clock_identity;
+                parent_ds.parent_port_identity.port_number = 0;
+                parent_ds.grandmaster_identity = defaultds.clock_identity;
+                parent_ds.grandmaster_clock_quality = defaultds.clock_quality;
+                parent_ds.grandmaster_priority_1 = defaultds.priority_1;
+                parent_ds.grandmaster_priority_2 = defaultds.priority_2;
+
+                time_properties_ds.leap59 = false;
+                time_properties_ds.leap61 = false;
```
[the remark this answers is not in the document]
Comment (davidv1992) [reply statime/src/port/mod.rs:299]: True, suggest we refactor this later (this was moved code)

File: statime/src/port/mod.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (davidv1992) [reply statime/src/port/mod.rs:300]: True, suggest we refactor this later (this was moved code)

File: statime/src/port/mod.rs [hunk 3]
```diff
@@ -236,15 +239,143 @@ impl Port<Running> {
...
+                let state = PortState::Slave(SlaveState::new(remote_master));
+
+                match &self.port_state {
+                    PortState::Listening | PortState::Master(_) | PortState::Passive => {
+                        self.set_forced_port_state(state);
+                        self.lifecycle.pending_action =
+                            Some(PortAction::ResetAnnounceReceiptTimer {
+                                duration: core::time::Duration::from_secs_f64(
+                                    2f64.powi(self.config.log_announce_interval as i32)
+                                        * (self.config.announce_receipt_timeout as f64),
```
[the remark this answers is not in the document]
Comment (davidv1992) [reply statime/src/port/mod.rs:346]: True, suggest we refactor this later (this was moved code)

**Review by bramtweedegolf — APPROVED** [review]

**Pull request merged**